### PR TITLE
Cythonize

### PR DESCRIPTION
**Pull request by loriab**
## Description
To ward off tampering with editable python files in distributed psi4, add build option to compile each python file into its own `.so`. The python loader, in its beneficence, doesn't care whether files are .py or .so, so long as they're in the right spot in the filesys.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] add `ENABLE_CYTHONIZE` build option to invoke
- [x] presently compiling most python files but not those that are a bother or repeat library names (qcdb, util), those that have references to missing objects, or the `__init__.py` files (need a better lib naming system), or the top-level py files (psi4/__init__.py, extras, header).
- [x] this is a crude cmake hookup. could be consolidated into a fn, and need to `rm -rf stage/lib/psi4/driver` to trigger total rebuild upon changing any py file.
- [x] corrected various missing imports and non-existing objects that compilation notices.

## Checklist
- [ ] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/driver/p4util/solvers.py [hunk 1]
```diff
@@ -555,6 +553,8 @@ def vector_dot(X, Y):
            The dot product  (X x Y)
         """
         pass
+    # cython doesn't like static+ decorators https://github.com/cython/cython/issues/1434#issuecomment-608975116
```
Comment (robertodr) [inline psi4/driver/p4util/solvers.py:556]: Nice catch!

File: psi4/driver/p4util/inpsight.py [hunk 2]
```diff
@@ -388,11 +388,11 @@ def position_camera(self):
 
         phi = math.pi*(-self.azimuth)/180.0
         theta = math.pi*(90.0 - self.elevation)/180.0
-        delta = [Rmax*cos(phi)*sin(theta), Rmax*sin(phi)*sin(theta), Rmax*cos(theta)]
+        delta = [Rmax*math.cos(phi)*math.sin(theta), Rmax*math.sin(phi)*math.sin(theta), Rmax*math.cos(theta)]
```
[the remark this answers is not in the document]
Comment (loriab) [reply psi4/driver/p4util/inpsight.py:391]: it's unaccessed code :-)

**Comment by loriab** [comment]
> The way I understand it, this will stop users from being able to tinker with the distributed versions of psi4 (eg. via conda), as they'll be in object files as opposed to source code. Do we really want that?

You understand correctly. And I agree, we don't want to prevent editing in general. No way will this be enabled by default for compile-yourself, and no changes are anticipated to the usual conda-package distributions either. This option is intended for the odd case of shipping off scientific calculations to gaming computers, where they might be tempted to take shortcuts. (Though if they can actually improve results by editing the driver, we'd better adopt them.)

File: psi4/driver/procrouting/dft/superfunctionals.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def build_superfunctional(name, restricted, npoints=None, deriv=1):
         if isinstance(sfunc, core.SuperFunctional):
             sup = (sfunc, False)
         # With Dispersion
-        elif isinstance(sup[0], core.SuperFunctional):
+        elif isinstance(sfunc[0], core.SuperFunctional):  # first arg was sup[0] but I guess this was never hit?
```
[the remark this answers is not in the document]
Comment (loriab) [reply psi4/driver/procrouting/dft/superfunctionals.py]: thanks for the test input! I've fixed it and added the test in the mp2 testing PR and here.

File: psi4/driver/p4util/inpsight.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (PeterKraus) [reply psi4/driver/p4util/inpsight.py:391]: Whatever even is this file, I probably don't want to know. Looks like a molecule renderer?

[the remark this answers is not in the document]
Comment (loriab) [reply psi4/driver/p4util/inpsight.py:391]: yes, prepares povray input. probably works very nicely (and programatically!) with a little attention.

**Pull request merged**